### PR TITLE
Refactor Hyrax::CollectionChangeSet terms methods and specs

### PR DESCRIPTION
**Pull request by bmquinn**
- Class methods for Hyrax::CollectionChangeSet terms methods (terms, primary, secondary)
- Fix specs for Hyrax::CollectionChangeSet

### CODE REVIEW

File: app/change_sets/hyrax/collection_change_set.rb [hunk 1]
```diff
@@ -36,26 +36,36 @@ def select_files
       Hash[all_files_with_access]
     end
 
-    # Terms that appear above the accordion
-    def primary_terms
-      [:title]
-    end
+    class << self
+      # Combined primary and secondary terms
+      def terms
```
Comment (jcoyne) [inline app/change_sets/hyrax/collection_change_set.rb]: what is this for? Why do we need it?

File: app/change_sets/hyrax/collection_change_set.rb [hunk 2]
```diff
@@ -36,26 +36,36 @@ def select_files
...
-       :publisher,
-       :date_created,
-       :subject,
-       :language,
-       :identifier,
-       :based_near,
-       :related_url,
-       :resource_type]
+      # Terms that appear above the accordion
+      def primary_terms
```
Comment (jcoyne) [inline app/change_sets/hyrax/collection_change_set.rb]: You just changed these instance methods to a class methods, why? That's going to mess up the views which are expecting to call this on the instance.

File: app/change_sets/hyrax/collection_change_set.rb (hunk 1 above)
Comment (bmquinn) [reply app/change_sets/hyrax/collection_change_set.rb]: Just trying to green up a failing test, the spec was failing on `described_class.terms` with a `NoMethodError` but we can get of it and re-work the test if needed.

File: app/change_sets/hyrax/collection_change_set.rb (hunk 2 above)
Comment (bmquinn) [reply app/change_sets/hyrax/collection_change_set.rb]: Ok thanks, will revert them.

File: app/change_sets/hyrax/collection_change_set.rb (hunk 1 above)
Comment (jcoyne) [reply app/change_sets/hyrax/collection_change_set.rb]: I believe that would be taken care of if you ported `autocreate_fields!` into CollectionsChangeSet, but I'm really not sure if `fields` has any utility anymore.

Comment (bmquinn) [reply app/change_sets/hyrax/collection_change_set.rb]: Ok cool, should we just merge this PR now to keep it moving and talk about `fields` with the group after standup tomorrow?

**Pull request merged**